cdap-security/src/main/java/co/cask/cdap/security/impersonation/ImpersonationInfo.java

```diff
@@ -19,18 +19,28 @@
 import com.google.common.base.Strings;
 
 import java.util.Objects;
+import javax.annotation.Nullable;
 
 /**
  * Encapsulates information necessary to impersonate a user - principal and keytab path.
  */
 public final class ImpersonationInfo {
   private final String principal;
   private final String keytabURI;
+  @Nullable
```

Comment:
Exposing null version is problematic. Why not have the NamespaceMetadata handle null and default it in the get method?